x-pack/solutions/security/plugins/security_solution/public/flyout/attack_details/hooks/use_find_attack_details_by_id/index.ts

```diff
@@ -0,0 +1,165 @@
+  : unknown;
+
+interface UseFindAttackDiscoveryById {
+  cancelRequest: () => void;
+  data: AttackDiscoveryItem | undefined;
+  error: unknown | undefined;
+  isLoading: boolean;
+  refetch: <TPageData>(
+    options?: (RefetchOptions & RefetchQueryFilters<TPageData>) | undefined
+  ) => Promise<QueryObserverResult<AttackDiscoveryInternalItem | undefined, unknown>>;
+  status: 'error' | 'idle' | 'loading' | 'success';
+}
+
+/**
+ * Fetch a single attack discovery by document id using the FIND endpoint.
+ * Uses the public/internal FIND route depending on the feature flag and
+ * maps the public API shape into the internal UI shape.
+ */
+export const useFindAttackDetailsById = ({
+  id,
+  http,
+  isAssistantEnabled,
+  refetchOnWindowFocus = false,
+}: Props): UseFindAttackDiscoveryById => {
+  const { addError } = useAppToasts();
+  const { attackDiscoveryPublicApiEnabled } = useKibanaFeatureFlags();
+  const abortController = useRef(new AbortController());
+
+  const cancelRequest = useCallback(() => {
+    abortController.current.abort();
+    abortController.current = new AbortController(); // LOCAL MUTATION
+  }, []);
+
+  const route = attackDiscoveryPublicApiEnabled
+    ? ATTACK_DISCOVERY_FIND
+    : ATTACK_DISCOVERY_INTERNAL_FIND;
+
+  const version = attackDiscoveryPublicApiEnabled
+    ? API_VERSIONS.public.v1
+    : API_VERSIONS.internal.v1;
+
+  const queryFn = useCallback(async () => {
+    const baseQuery = {
+      ids: [id],
+      page: 1,
+      per_page: 1,
+    };
+
+    if (attackDiscoveryPublicApiEnabled) {
+      return http.fetch<AttackDiscoveryFindResponse>(route, {
+        method: 'GET',
+        version,
+        query: {
+          ...baseQuery,
+          enable_field_rendering: true, // always true to enable rendering fields using the `{{ user.name james }}` syntax
+          with_replacements: false, // always false because Attack discoveries rendered in Kibana may be passed as context to a conversation, and to enable the user to see the original alert details via the `Show anonymized values` toggle
+        },
+        signal: abortController.current.signal,
+      });
+    }
+
+    return http.fetch<AttackDiscoveryFindInternalResponse>(route, {
+      method: 'GET',
+      version,
+      query: baseQuery,
+      signal: abortController.current.signal,
+    });
+  }, [http, route, version, id, attackDiscoveryPublicApiEnabled]);
+
+  const { data, error, isLoading, refetch, status } = useQuery(
+    ['GET', route, id, isAssistantEnabled],
+    queryFn,
+    {
+      enabled: isAssistantEnabled,
+      // Transform the API response's data items into UI-friendly alerts
+      // only when the public API is enabled. Otherwise return the raw
+      // response shape (internal API uses different field names).
+      select: (
+        response: AttackDiscoveryFindResponse | AttackDiscoveryFindInternalResponse
+      ): AttackDiscoveryInternalItem => {
+        if (attackDiscoveryPublicApiEnabled) {
```

Comment:
I know you got that code from another place in the attack discovery page, but it would be nice to understand why we need this?